src/XamlStyler/DocumentProcessors/ElementDocumentProcessor.cs

```diff
@@ -319,9 +320,30 @@ private int AttributeInfoComparison(AttributeInfo x, AttributeInfo y)
                 return x.OrderRule.Priority.CompareTo(y.OrderRule.Priority);
             }
 
-            return this.options.OrderAttributesByName
-                ? String.Compare(x.Name, y.Name, StringComparison.Ordinal)
-                : 0;
+            if (this.options.OrderAttributesByName)
+            {
+                if (x.AttributeHasIgnoredNamespace && y.AttributeHasIgnoredNamespace)
+                {
+                    return String.Compare(x.AttributeNameWithoutNamespace, y.AttributeNameWithoutNamespace, StringComparison.Ordinal);
+                }
+                // If we have attribute with ignored namespace, we want to compare it by full name
+                // if it is compared with analogical attribute without this namespace.
+                else if (x.AttributeHasIgnoredNamespace && x.AttributeNameWithoutNamespace != y.Name)
```

Comment:
I think we want to use String.Equals() or String.Compare() here (and below) too.

Reply:
corrected